index.js

```diff
@@ -163,6 +71,156 @@ class ShareSheet extends React.Component<Props> {
   }
 }
 
+type Options = {
+  url: string,
+  urls: Array<string>,
+  type: string,
+  message: string,
+  title?: string,
+  subject?: string,
+  excludedActivityTypes?: string,
+  failOnCancel?: boolean,
+  showAppsToView?: boolean,
+};
+type OpenReturn = { app?: string, dismissedAction?: boolean };
+type ShareSingleReturn = { message: string };
+
+const requireAndAskPermissions = (options: Options): Promise<any> => {
+  if ((options.url || options.urls) && Platform.OS === 'android') {
+    return new Promise((resolve, reject) => {
+      const urls = options.urls || [options.url];
+      Promise.all(
+        urls.map(url => {
+          return new Promise((res, rej) => {
+            NativeModules.RNShare.isBase64File(
```

Comment:
Why didn't  you make `isBase64File` return a promise? It wouldn't be necessary to create a promise with `new Promise`

Reply:
Since all other methods were written using callback, so I just wanted to avoid the discrepancies.